### DISCUSSION TITLE
Best way to connect to database with multiple server routes?

### DISCUSSION THREAD

**Discussion by stenuto** [nuxt/nuxt#16260] [Questions]
I'm using `/api` server routes and [pg-promise](https://github.com/vitaly-t/pg-promise) to create various endpoints for my application. I'd like to reuse the database object below for various endpoints I'm creating.

```
import pgPromise from "pg-promise";
const pgp = pgPromise({
  /* Initialization Options */
});
const db = pgp(process.env.CONNECTION_STRING);
```

I've tried adding `module.exports = { db, pgp }` and then importing that in `/api/my-endpoint.js` like so:
`import * as db from '../utils/db';`

...but `db` is undefined.

What is the best way to do this? Can server middleware be utilized here?

**Comment by wobsoriano** [comment]
Looks to be working on my side with this simple config:

```ts
// lib/db.ts
import pgPromise from 'pg-promise'

export const pgp = pgPromise({})

export const db = pgp('')
```

```ts
// server/api/my-endpoint.ts
import * as db from '~~/lib/pg'

export default eventHandler((event) => {
  console.log(db) // works
  return { ok: true }
})
```
